skiko/src/commonMain/kotlin/org/jetbrains/skia/paragraph/TypefaceFontProvider.kt

```diff
@@ -12,22 +12,40 @@ import org.jetbrains.skia.impl.NativePointer
 import org.jetbrains.skia.impl.getPtr
 import org.jetbrains.skia.impl.interopScope
 
-class TypefaceFontProvider : FontMgr(TypefaceFontProvider_nMake()) {
+class TypefaceFontProvider private constructor(
+    ptr: NativePointer,
+    internal val isFallbackProvider: Boolean
+) : FontMgr(ptr) {
+
+    constructor(): this(TypefaceFontProvider_nMake(), false)
+
     companion object {
         init {
             staticLoad()
         }
+
+        fun createAsFallbackProvider(): TypefaceFontProvider {
```

Comment:
It looks like there is no reason for user to not always create `TypefaceFontProvider.createAsFallbackProvider`. The API and behaviour is the same, the only difference is that the fallback one can be passed to another function.

We can just pass `TypefaceFontProvider_nMakeAsFallbackProvider` in the constructor, and remove this function (and other unused functions/type checks after that).